src/core/types/keys/key_agreement/private_key.rs

```diff
@@ -0,0 +1,189 @@
+use crate::prelude::*;
+use crypto::keys::x25519::SecretKey as X25519PrivateKey;
+use crypto::keys::x25519::SharedSecret;
+
+#[derive(derive_more::Debug)]
+#[debug("{}", self.to_hex())]
+pub struct KeyAgreementPrivateKey(X25519PrivateKey);
```

Comment:
missing doc, maybe `/// Private Key on Curve25519 used for key agreement (ECDH).`